cmake/toolchain/xtools/target.cmake

```diff
@@ -13,6 +13,8 @@ set(CROSS_COMPILE_TARGET_sparc     sparc-zephyr-elf)
 set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
 set(SYSROOT_TARGET       ${CROSS_COMPILE_TARGET})
 
+set(ZEPHYR_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE y)
```

Comment:
this doesn't set the variable in environment.

also this commit brought my attention to this PR: #45656

We should not in Kconfig rely directly on environment settings define by the user, as it can result in incremental builds starting to behave differently.

Instead such knowledge should come from the toolchains Kconfig like here: 
https://github.com/zephyrproject-rtos/zephyr/blob/1dc69c064821d3d37f80b10c7ed15645ea543fce/cmake/toolchain/gnuarmemb/Kconfig#L8

environment variables to be used inside Kconfig should be exported by CMake kconfig integration here:
https://github.com/zephyrproject-rtos/zephyr/blob/1dc69c064821d3d37f80b10c7ed15645ea543fce/cmake/modules/kconfig.cmake#L110

but for the TLS, I believe the first approachh would be best.
Done correctly, it could also provide users an option to state that a toolchain doesn't support TLS.